bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,319 @@
+                return error_embed
+
+            soup = BeautifulSoup(await response.text(), features="lxml")
+            first_kata_div = await to_thread(soup.find_all, "div", class_="item-title px-0")
+
+            if not first_kata_div:
+                raise commands.BadArgument("No katas could be found with the filters provided.")
+            elif len(first_kata_div) >= 3:
+                first_kata_div = choice(first_kata_div[:3])
+            elif 'q=' not in search_link:
+                first_kata_div = choice(first_kata_div)
+            else:
+                first_kata_div = first_kata_div[0]
+
+            # There are numerous divs before arriving at the id of the kata, which can be used for the link.
+            first_kata_id = first_kata_div.a["href"].split("/")[-1]
+            return first_kata_id
+
+    async def kata_information(self, kata_id: str) -> Union[dict, Embed]:
+        """
+        Returns the information about the Kata.
+
+        Uses the codewars.com API to get information about the kata using `kata_id`.
+        """
+        async with self.bot.http_session.get(API_ROOT.format(kata_id=kata_id)) as response:
+            if response.status != 200:
+                error_embed = Embed(
+                    title=choice(NEGATIVE_REPLIES),
+                    description="We ran into an error when getting the kata information, try again later.",
+                    color=Colours.soft_red
+                )
+                logger.error(f"Unexpected response from codewars.com/api/v1, status code: {response.status}")
+                return error_embed
+
+            return await response.json()
+
+    @staticmethod
+    def main_embed(kata_information: dict) -> Embed:
+        """Creates the main embed which displays the name, difficulty and description of the kata."""
+        kata_description = kata_information["description"]
+        kata_url = f"https://codewars.com/kata/{kata_information['id']}"
+
+        # Ensuring it isn't over the length 1024
+        if len(kata_description) > 1024:
+            kata_description = "\n".join(kata_description[:1000].split("\n")[:-1]) + "..."
+            kata_description += f"\n[Read more...]({kata_url})"
+
+        kata_embed = Embed(
+            title=kata_information['name'],
+            description=kata_description,
+            color=MAPPING_OF_KYU[int(kata_information['rank']['name'].replace(' kyu', ''))],
+            url=kata_url
+        )
+        kata_embed.add_field(name="Difficulty", value=kata_information['rank']['name'], inline=False)
+        return kata_embed
+
+    @staticmethod
+    def language_embed(kata_information: dict) -> Embed:
+        """Creates the 'language embed' which displays all languages the kata supports."""
+        languages = '\n'.join(map(str.title, kata_information['languages']))
+        language_embed = Embed(
+            title="Supported Languages",
```

Comment:
Can the title for these include the title for the Kata the same way the main embed does?

Maybe in the style of "Main Information | Name on Billboard" (or the other way around) or in some other style if you prefer that. I got a bit lost after selecting other information.

*Applies to the other embeds*

Reply:
How does this look?
![image](https://user-images.githubusercontent.com/82843611/134756692-95516cff-5690-4d7c-afd6-c50e7efc3274.png)

Reply:
I think it is good there in the select menu, I am talking about the embed title.

The issue is that the main information has the title, but for example "Programming Languages" has exactly that as its title (not the name of the Kata included somewhere).

Reply:
Ah, alright